test/strict.coffee

```diff
@@ -17,8 +17,10 @@
 # * `eval` or `arguments` as the operand of a post/pre-fix inc/dec-rement expression
 
 # helper to assert that code complies with strict prohibitions
-strict = (code, msg) ->
+strict = (code, msg, ast) ->
```

Comment:
> The idea is fine, but I’m not a fan of mixing styles. We can refactor to move to options arguments later

Ok first of all in looking more closely at this I realized that it wasn't actually triggering the `ast` versions of the tests, and it's very related to options-style arguments:

I was making calls like `strict "#{keyword}++", ast: yes`, but with the existing signature, `ast: yes` was being treated as the `msg` argument. So if we just add a sequential argument for `ast`, it'll be called like
```
strict "#{keyword}++", null, yes
```
That's pretty ugly and you basically have to go back and read the definition of `strict()` to know how to interpret the arguments

I know that you'd rather postpone these types of things until later refactors, so I updated it to use the above sequential calling style for now

For when the time comes, let me make an argument against "mixing styles" (always) being a bad thing:

I'm understanding "mixing styles" to mean using a combination of sequential and options-object arguments

To me, it's too inflexible to say "always all sequential arguments or just an options argument". I think the determining factor should be what the "natural" style is to call the function. Often times the first (couple) arguments are self-evident and required so it's more awkward to label them as options than just to supply them as sequential arguments

Take `CoffeeScript.compile()` as an existing example of "mixed style". You call it like eg
```
CoffeeScript.compile code, ast: yes
```
To insist on non-mixed style, you'd have to change the signature to eg
```
CoffeeScript.compile {code, ast: yes}
```
This isn't bad (especially with the succintness of object shorthand), it's far preferable to a fully sequential calling style. But I think there's a reason this API was designed the way it was, `code` is a rather obvious first argument to `compile()` so doesn't particularly benefit from being an option arg. But then everything after it should definitely be an option

Again (thinking of the discussion about trailing implicit option object calling style), I think Ruby is perhaps the strongest influence on Coffeescript "style" in general and calling styles in particular with its emphasis on natural, uncluttered readability. And in Ruby an implicit call with trailing "implicit" options (after possibly one or more sequential arguments) would be a typical style afaik. I feel like that community would say "choose the calling interface that makes calling it easiest and most readable" rather than something as dogmatic as "all sequential or all options, always"

Reply:
Yeah maybe sequential here is being too dogmatic. Though I think the original authors would instead make a new helper like `strictAst` that could have the same signature as `strict`.